web/generic_scripting.js

```diff
@@ -56,7 +56,7 @@ class GenericScripting {
 
   async dispatchEventInSandbox(event) {
     const sandbox = await this._ready;
-    sandbox.dispatchEvent(event);
+    setTimeout(() => sandbox.dispatchEvent(event), 0);
```

Comment:
Was this change necessary to reduce intermittent failures, or just left-over from previous debugging?

Reply:
I added it to reduce intermittent failures.
Locally, I had some issues with chrome integration tests, and the first fix I made was to dispatch the event (in the `beforeinput` listener) using a setTimeout. After thinking about that, in Firefox, the event is dispatched in the sandbox asynchronously, so I thought that it wouldn't hurt to have same behaviour in the tests.

Reply:
OK, that was just a bit surprising since *naively* I'd have assumed that the previous line (where we await a Promise) already introduces asynchronicity in the event dispatching.

Anyway, if we're adding this additional "delay" to the event dispatching there's *potentially* a slightly larger risk than before that we're now attempting to dispatch an event after destruction have started.
Given all the various `dispatchEventInSandbox` call-sites it's difficult to easily tell what an error could lead to, so how about also changing
https://github.com/mozilla/pdf.js/blob/23b6fde9fc92b827f054e21978e32ce9c13dab82/src/pdf.sandbox.js#L106
into the following to prevent any future errors elsewhere:
```javascript
this.support?.callSandboxFunction("dispatchEvent", event); 
```